### PR TITLE
Add Typespec API definitions

### PR DESCRIPTION
**Pull request by munishchouhan**
This PR will add documentation based on typespec

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,6 +108,14 @@ container registry where the image is stored, while the instrumented layers are
...
     ```
 
+## TypeSpec API Specifications
+
+    You can find the API specifications using (typespec)[https://github.com/microsoft/typespec] in typespec directory. Use following command to generate the API specifications.
+
+    ```bash
+    tsp compile typespec
+    ```
+    Check `tsp-output` directory for the generated API specifications.
```
Comment (ewels) [inline README.md:118]:
````suggestion
You can find the API specifications using (typespec)[https://github.com/microsoft/typespec] in typespec directory. Use following command to generate the API specifications.

```bash
tsp compile typespec
```

Check `tsp-output` directory for the generated API specifications.
````

Indentation here made the markdown render as a code block.

**Comment by pditommaso** [comment]
Tried this and got many errors.

```
» tsp compile typespec
TypeSpec compiler v0.55.0

Diagnostics were reported during compilation:

/Users/pditommaso/Projects/wave/typespec/main.tsp:1:1 - error import-not-found: Couldn't resolve import "@typespec/http"
> 1 | import "@typespec/http";
    | ^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/main.tsp:2:1 - error import-not-found: Couldn't resolve import "@typespec/rest"
> 2 | import "@typespec/rest";
    | ^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/main.tsp:3:1 - error import-not-found: Couldn't resolve import "@typespec/openapi3"
> 3 | import "@typespec/openapi3";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/models/models.tsp:5:1 - error import-not-found: Couldn't resolve import "./ContainerInspectResponse.tsp"
> 5 | import "./ContainerInspectResponse.tsp";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error import-not-found: Couldn't resolve import "@typespec/openapi3"
/Users/pditommaso/Projects/wave/typespec/routes.tsp:3:16 - error invalid-ref: Namespace TypeSpec doesn't have member Http
> 3 | using TypeSpec.Http;
    |                ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:4:16 - error invalid-ref: Namespace TypeSpec doesn't have member Rest
> 4 | using TypeSpec.Rest;
    |                ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:9:2 - error unknown-identifier: Unknown identifier server
> 9 | @server("https://wave.seqera.io", "wave endopint")
    |  ^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:9:1 - error unknown-decorator: Unknown decorator
> 9 | @server("https://wave.seqera.io", "wave endopint")
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:11:4 - error unknown-identifier: Unknown identifier route
> 11 |   @route("/v1alpha2/container")
     |    ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:11:3 - error unknown-decorator: Unknown decorator
> 11 |   @route("/v1alpha2/container")
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:14:39 - error unknown-identifier: Unknown identifier body
> 14 |     @post op createV1Alpha2Container(@body requestBody: ContainerRequest): {
     |                                       ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:14:38 - error unknown-decorator: Unknown decorator
> 14 |     @post op createV1Alpha2Container(@body requestBody: ContainerRequest): {
     |                                      ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:15:6 - error unknown-identifier: Unknown identifier body
> 15 |     @body response: ContainerResponse;
     |      ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:15:5 - error unknown-decorator: Unknown decorator
> 15 |     @body response: ContainerResponse;
     |     ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:16:6 - error unknown-identifier: Unknown identifier statusCode
> 16 |     @statusCode statusCode: 200;
     |      ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:16:5 - error unknown-decorator: Unknown decorator
> 16 |     @statusCode statusCode: 200;
     |     ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:14:6 - error unknown-identifier: Unknown identifier post
> 14 |     @post op createV1Alpha2Container(@body requestBody: ContainerRequest): {
     |      ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:14:5 - error unknown-decorator: Unknown decorator
> 14 |     @post op createV1Alpha2Container(@body requestBody: ContainerRequest): {
     |     ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:21:4 - error unknown-identifier: Unknown identifier route
> 21 |   @route("/v1alpha1/builds/{buildId}")
     |    ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:21:3 - error unknown-decorator: Unknown decorator
> 21 |   @route("/v1alpha1/builds/{buildId}")
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:24:29 - error unknown-identifier: Unknown identifier path
> 24 |     @get op getBuildRecord(@path buildId: string): {
     |                             ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:24:28 - error unknown-decorator: Unknown decorator
> 24 |     @get op getBuildRecord(@path buildId: string): {
     |                            ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:25:8 - error unknown-identifier: Unknown identifier body
> 25 |       @body response: WaveBuildRecord;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:25:7 - error unknown-decorator: Unknown decorator
> 25 |       @body response: WaveBuildRecord;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:26:8 - error unknown-identifier: Unknown identifier statusCode
> 26 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:26:7 - error unknown-decorator: Unknown decorator
> 26 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:28:8 - error unknown-identifier: Unknown identifier statusCode
> 28 |       @statusCode statusCode: 404;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:28:7 - error unknown-decorator: Unknown decorator
> 28 |       @statusCode statusCode: 404;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:24:6 - error unknown-identifier: Unknown identifier get
> 24 |     @get op getBuildRecord(@path buildId: string): {
     |      ^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:24:5 - error unknown-decorator: Unknown decorator
> 24 |     @get op getBuildRecord(@path buildId: string): {
     |     ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:32:29 - error unknown-identifier: Unknown identifier path
> 32 |     @get op getBuildStatus(@path buildId: string): {
     |                             ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:32:28 - error unknown-decorator: Unknown decorator
> 32 |     @get op getBuildStatus(@path buildId: string): {
     |                            ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:33:8 - error unknown-identifier: Unknown identifier body
> 33 |       @body response: BuildStatusResponse;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:33:7 - error unknown-decorator: Unknown decorator
> 33 |       @body response: BuildStatusResponse;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:34:8 - error unknown-identifier: Unknown identifier statusCode
> 34 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:34:7 - error unknown-decorator: Unknown decorator
> 34 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:36:8 - error unknown-identifier: Unknown identifier statusCode
> 36 |       @statusCode statusCode: 404;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:36:7 - error unknown-decorator: Unknown decorator
> 36 |       @statusCode statusCode: 404;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:31:6 - error unknown-identifier: Unknown identifier route
> 31 |     @route("/status")
     |      ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:31:5 - error unknown-decorator: Unknown decorator
> 31 |     @route("/status")
     |     ^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:32:6 - error unknown-identifier: Unknown identifier get
> 32 |     @get op getBuildStatus(@path buildId: string): {
     |      ^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:32:5 - error unknown-decorator: Unknown decorator
> 32 |     @get op getBuildStatus(@path buildId: string): {
     |     ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:40:27 - error unknown-identifier: Unknown identifier path
> 40 |     @get op getBuildLogs(@path buildId: string): {
     |                           ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:40:26 - error unknown-decorator: Unknown decorator
> 40 |     @get op getBuildLogs(@path buildId: string): {
     |                          ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:41:8 - error unknown-identifier: Unknown identifier body
> 41 |       @body response: string;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:41:7 - error unknown-decorator: Unknown decorator
> 41 |       @body response: string;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:42:8 - error unknown-identifier: Unknown identifier statusCode
> 42 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:42:7 - error unknown-decorator: Unknown decorator
> 42 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:44:8 - error unknown-identifier: Unknown identifier statusCode
> 44 |       @statusCode statusCode: 404;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:44:7 - error unknown-decorator: Unknown decorator
> 44 |       @statusCode statusCode: 404;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:39:6 - error unknown-identifier: Unknown identifier route
> 39 |     @route("/logs")
     |      ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:39:5 - error unknown-decorator: Unknown decorator
> 39 |     @route("/logs")
     |     ^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:40:6 - error unknown-identifier: Unknown identifier get
> 40 |     @get op getBuildLogs(@path buildId: string): {
     |      ^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:40:5 - error unknown-decorator: Unknown decorator
> 40 |     @get op getBuildLogs(@path buildId: string): {
     |     ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:49:4 - error unknown-identifier: Unknown identifier route
> 49 |   @route("/v1alpha1/scans/{scanId}")
     |    ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:49:3 - error unknown-decorator: Unknown decorator
> 49 |   @route("/v1alpha1/scans/{scanId}")
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:52:24 - error unknown-identifier: Unknown identifier path
> 52 |     @get op scanImage(@path scanId: string) : { 
     |                        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:52:23 - error unknown-decorator: Unknown decorator
> 52 |     @get op scanImage(@path scanId: string) : { 
     |                       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:53:8 - error unknown-identifier: Unknown identifier body
> 53 |       @body response: WaveScanRecord;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:53:7 - error unknown-decorator: Unknown decorator
> 53 |       @body response: WaveScanRecord;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:54:8 - error unknown-identifier: Unknown identifier statusCode
> 54 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:54:7 - error unknown-decorator: Unknown decorator
> 54 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:56:8 - error unknown-identifier: Unknown identifier statusCode
> 56 |       @statusCode statusCode: 404;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:56:7 - error unknown-decorator: Unknown decorator
> 56 |       @statusCode statusCode: 404;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:52:6 - error unknown-identifier: Unknown identifier get
> 52 |     @get op scanImage(@path scanId: string) : { 
     |      ^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:52:5 - error unknown-decorator: Unknown decorator
> 52 |     @get op scanImage(@path scanId: string) : { 
     |     ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:61:4 - error unknown-identifier: Unknown identifier route
> 61 |   @route("/v1alpha1/inspect")
     |    ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:61:3 - error unknown-decorator: Unknown decorator
> 61 |   @route("/v1alpha1/inspect")
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:64:32 - error unknown-identifier: Unknown identifier body
> 64 |     @post op inspectContainer(@body requestBody: ContainerInspectRequest): {
     |                                ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:64:31 - error unknown-decorator: Unknown decorator
> 64 |     @post op inspectContainer(@body requestBody: ContainerInspectRequest): {
     |                               ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:65:23 - error unknown-identifier: Unknown identifier ContainerInspectResponse
> 65 |       @body response: ContainerInspectResponse;
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:65:8 - error unknown-identifier: Unknown identifier body
> 65 |       @body response: ContainerInspectResponse;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:65:7 - error unknown-decorator: Unknown decorator
> 65 |       @body response: ContainerInspectResponse;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:66:8 - error unknown-identifier: Unknown identifier statusCode
> 66 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:66:7 - error unknown-decorator: Unknown decorator
> 66 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:68:8 - error unknown-identifier: Unknown identifier statusCode
> 68 |       @statusCode statusCode: 404;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:68:7 - error unknown-decorator: Unknown decorator
> 68 |       @statusCode statusCode: 404;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:64:6 - error unknown-identifier: Unknown identifier post
> 64 |     @post op inspectContainer(@body requestBody: ContainerInspectRequest): {
     |      ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:64:5 - error unknown-decorator: Unknown decorator
> 64 |     @post op inspectContainer(@body requestBody: ContainerInspectRequest): {
     |     ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:73:4 - error unknown-identifier: Unknown identifier route
> 73 |   @route("/v1alpha2/metrics")
     |    ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:73:3 - error unknown-decorator: Unknown decorator
> 73 |   @route("/v1alpha2/metrics")
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:77:30 - error unknown-identifier: Unknown identifier query
> 77 |     @get op getBuildMetrics(@query date?: string, @query org?: string): {
     |                              ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:77:29 - error unknown-decorator: Unknown decorator
> 77 |     @get op getBuildMetrics(@query date?: string, @query org?: string): {
     |                             ^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:77:52 - error unknown-identifier: Unknown identifier query
> 77 |     @get op getBuildMetrics(@query date?: string, @query org?: string): {
     |                                                    ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:77:51 - error unknown-decorator: Unknown decorator
> 77 |     @get op getBuildMetrics(@query date?: string, @query org?: string): {
     |                                                   ^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:78:8 - error unknown-identifier: Unknown identifier body
> 78 |       @body response: MetricsResponse;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:78:7 - error unknown-decorator: Unknown decorator
> 78 |       @body response: MetricsResponse;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:79:8 - error unknown-identifier: Unknown identifier statusCode
> 79 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:79:7 - error unknown-decorator: Unknown decorator
> 79 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:76:6 - error unknown-identifier: Unknown identifier route
> 76 |     @route("/builds")
     |      ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:76:5 - error unknown-decorator: Unknown decorator
> 76 |     @route("/builds")
     |     ^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:77:6 - error unknown-identifier: Unknown identifier get
> 77 |     @get op getBuildMetrics(@query date?: string, @query org?: string): {
     |      ^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:77:5 - error unknown-decorator: Unknown decorator
> 77 |     @get op getBuildMetrics(@query date?: string, @query org?: string): {
     |     ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:83:29 - error unknown-identifier: Unknown identifier query
> 83 |     @get op getPullMetrics(@query date?: string, @query org?: string):  {
     |                             ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:83:28 - error unknown-decorator: Unknown decorator
> 83 |     @get op getPullMetrics(@query date?: string, @query org?: string):  {
     |                            ^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:83:51 - error unknown-identifier: Unknown identifier query
> 83 |     @get op getPullMetrics(@query date?: string, @query org?: string):  {
     |                                                   ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:83:50 - error unknown-decorator: Unknown decorator
> 83 |     @get op getPullMetrics(@query date?: string, @query org?: string):  {
     |                                                  ^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:84:8 - error unknown-identifier: Unknown identifier body
> 84 |       @body response: MetricsResponse;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:84:7 - error unknown-decorator: Unknown decorator
> 84 |       @body response: MetricsResponse;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:85:8 - error unknown-identifier: Unknown identifier statusCode
> 85 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:85:7 - error unknown-decorator: Unknown decorator
> 85 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:82:6 - error unknown-identifier: Unknown identifier route
> 82 |     @route("/pulls")
     |      ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:82:5 - error unknown-decorator: Unknown decorator
> 82 |     @route("/pulls")
     |     ^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:83:6 - error unknown-identifier: Unknown identifier get
> 83 |     @get op getPullMetrics(@query date?: string, @query org?: string):  {
     |      ^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:83:5 - error unknown-decorator: Unknown decorator
> 83 |     @get op getPullMetrics(@query date?: string, @query org?: string):  {
     |     ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:89:35 - error unknown-identifier: Unknown identifier query
> 89 |     @get op getFusionPullMetrics(@query date?: string, @query org?: string):  {
     |                                   ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:89:34 - error unknown-decorator: Unknown decorator
> 89 |     @get op getFusionPullMetrics(@query date?: string, @query org?: string):  {
     |                                  ^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:89:57 - error unknown-identifier: Unknown identifier query
> 89 |     @get op getFusionPullMetrics(@query date?: string, @query org?: string):  {
     |                                                         ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:89:56 - error unknown-decorator: Unknown decorator
> 89 |     @get op getFusionPullMetrics(@query date?: string, @query org?: string):  {
     |                                                        ^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:90:8 - error unknown-identifier: Unknown identifier body
> 90 |       @body response: MetricsResponse;
     |        ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:90:7 - error unknown-decorator: Unknown decorator
> 90 |       @body response: MetricsResponse;
     |       ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:91:8 - error unknown-identifier: Unknown identifier statusCode
> 91 |       @statusCode statusCode: 200;
     |        ^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:91:7 - error unknown-decorator: Unknown decorator
> 91 |       @statusCode statusCode: 200;
     |       ^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:88:6 - error unknown-identifier: Unknown identifier route
> 88 |     @route("/fusion/pulls")
     |      ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:88:5 - error unknown-decorator: Unknown decorator
> 88 |     @route("/fusion/pulls")
     |     ^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:89:6 - error unknown-identifier: Unknown identifier get
> 89 |     @get op getFusionPullMetrics(@query date?: string, @query org?: string):  {
     |      ^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:89:5 - error unknown-decorator: Unknown decorator
> 89 |     @get op getFusionPullMetrics(@query date?: string, @query org?: string):  {
     |     ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:96:29 - error unknown-identifier: Unknown identifier body
> 96 |     @post op validateCreds(@body request: ValidateRegistryCredsRequest):  boolean;
     |                             ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:96:28 - error unknown-decorator: Unknown decorator
> 96 |     @post op validateCreds(@body request: ValidateRegistryCredsRequest):  boolean;
     |                            ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:95:4 - error unknown-identifier: Unknown identifier route
> 95 |   @route("validate-creds")
     |    ^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:95:3 - error unknown-decorator: Unknown decorator
> 95 |   @route("validate-creds")
     |   ^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:96:6 - error unknown-identifier: Unknown identifier post
> 96 |     @post op validateCreds(@body request: ValidateRegistryCredsRequest):  boolean;
     |      ^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:96:5 - error unknown-decorator: Unknown decorator
> 96 |     @post op validateCreds(@body request: ValidateRegistryCredsRequest):  boolean;
     |     ^^^^^

Found 124 errors.
```

**Comment by munishchouhan** [comment]
@pditommaso please run these commands, I have updated the docs too
```
cd typespec

tsp install

tsp compile .
```

**Comment by pditommaso** [comment]
Much better, only to errors 😄 

```
» tsp compile .
TypeSpec compiler v0.57.0

Diagnostics were reported during compilation:

/Users/pditommaso/Projects/wave/typespec/models/models.tsp:5:1 - error import-not-found: Couldn't resolve import "./ContainerInspectResponse.tsp"
> 5 | import "./ContainerInspectResponse.tsp";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/Users/pditommaso/Projects/wave/typespec/routes.tsp:65:23 - error unknown-identifier: Unknown identifier ContainerInspectResponse
> 65 |       @body response: ContainerInspectResponse;
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^

Found 2 errors.

```

**Comment by munishchouhan** [comment]
@pditommaso please  `git pull` and try again

**Comment by pditommaso** [comment]
Good ✌️

File: typespec/models/BuildStatusResponse.tsp
Comment (pditommaso) [inline typespec/models/BuildStatusResponse.tsp:1]: any problem moving to `model` instead of `models` ?

Comment (munishchouhan) [reply typespec/models/BuildStatusResponse.tsp:1]: here, they have used models, so I did the same 
https://typespec.io/docs/language-basics/imports

Comment (pditommaso) [reply typespec/models/BuildStatusResponse.tsp:1]: fair enough

**Comment by munishchouhan** [comment]
> To close this, think we need to add a Github action that checks the model is correct, likely running `tsp compile .`

ok i will add that

**Comment by munishchouhan** [comment]
getting rate limit reached error from docker hub, will try and run tests tomorrow
```
Assertion failed: 

assert testcontainers0.execInContainer("docker","pull","hello-world").exitCode==0
                       |                                              |       |
                       |                                              1       false
                       Container.ExecResult(exitCode=1, stdout=Using default tag: latest
                       latest: Pulling from library/hello-world
                       , stderr=toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
                       )

	at io.seqera.wave.test.SecureDockerRegistryContainer$Trait$Helper.initRegistryContainer(SecureDockerRegistryContainer.groovy:56)
	at io.seqera.wave.auth.RegistryAuthServiceTest.setupSpec(RegistryAuthServiceTest.groovy:71)
```

**Comment by munishchouhan** [comment]
@pditommaso added validation step in CI

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -72,6 +72,21 @@ jobs:
...
+      - name: Install tsp
+        run: npm install -g @typespec/compiler
+
+      - name: Validate tsp files
+        run: |
+          cd typespec
+          tsp install
+          tsp compile .
+          cd ..
+
```
Comment (pditommaso) [inline .github/workflows/build.yml:89]: let's move into its own workflow, like codespell for example

Comment (munishchouhan) [reply .github/workflows/build.yml]: done

Comment (pditommaso) [reply .github/workflows/build.yml]: Excellent

**Pull request merged**